app/components/CrashDiagramCard.tsx

```diff
@@ -2,63 +2,161 @@ import { useState } from "react";
+            <span>Zoom Out</span>
+          </AlignedLabel>
+        </Button>
+      </ButtonGroup>
+      <Button size="sm" variant="primary" onClick={handleReset} title="Reset">
+        <AlignedLabel>
+          <SlActionUndo className="me-2" />
+          <span>Reset</span>
+        </AlignedLabel>
+      </Button>
+    </div>
+  );
+};
+
+const RotateControls = ({
+  rotation,
+  setRotation,
+}: {
+  rotation: number;
+  setRotation: (value: number) => void;
+}) => {
+  const rotate = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRotation(Number(event.target.value));
+  };
+
+  return (
+    <div className="mt-2">
+      <Form.Range
+        min="-180"
+        max="180"
+        value={rotation}
+        id="formControlRange"
+        onChange={rotate}
+        style={{ width: "70%" }}
+        title="Rotate Diagram"
+      />
+    </div>
+  );
+};
+
 export default function CrashDiagramCard({ crash }: { crash: Crash }) {
   const [diagramError, setDiagramError] = useState(false);
+  const [rotation, setRotation] = useState(0);
 
   return (
     <Card>
       <Card.Header>Diagram</Card.Header>
       <Card.Body className="crash-header-card-body text-center">
-        {!diagramError && (
-          <Image
-            fluid
-            style={{
-              height: "100%",
-              maxWidth: "100%",
-            }}
-            src={`${CR3_DIAGRAM_BASE_URL}/${crash.record_locator}.jpeg`}
-            alt="crash diagram"
-            onError={() => {
-              console.error("Error loading CR3 diagram image");
-              setDiagramError(true);
-            }}
-          />
-        )}
-        {diagramError && crash.is_temp_record && (
-          <Alert variant="info">
-            <i className="fa fa-info-circle" /> Crash diagrams are not available
-            for user-created crash records
-          </Alert>
-        )}
-        {diagramError && !crash.is_temp_record && (
-          <Alert variant="danger">
-            <p>
-              The crash diagram is not available. Typically, this indicates
-              there was an error when processing this crash&aposs CR3 PDF.
-            </p>
-            <p>
-              For additional assistance, you can&nbsp;
-              <a
-                href="https://atd.knack.com/dts#new-service-request/?view_249_vars=%7B%22field_398%22%3A%22Bug%20Report%20%E2%80%94%20Something%20is%20not%20working%22%2C%22field_399%22%3A%22Vision%20Zero%20Editor%22%7D"
-                target="_blank"
-                rel="noopener noreferrer"
-              >
-                report a bug
-                <i className="fa fa-external-link"></i>
-              </a>
-              .
-            </p>
-          </Alert>
-        )}
+        <TransformWrapper initialScale={1}>
+          {!diagramError && <ZoomResetControls setRotation={setRotation} />}
+          <TransformComponent>
+            <div style={{ height: "330px", width: "100%", overflow: "hidden" }}>
```

Comment:
Frank, this is looking better, but I think we can get it to a place where we're not hardcoding any `div` sizes. 

Looking at the html markup, I think we can probably nix the below div completely, sine the `TransformWrapper` and `TransformComponent` are adding their own divs.

Here's a quick demo of leaning into the flex utilities for this. By making the `Card.Body` a flex-column, the rest of the divs should stack up without overflowing. I think you can also strip out the height/width on the `Image` component as well—since the `fluid` prop is going to let it fill the parent container.

```jsx
   <Card.Body className="crash-header-card-body text-center d-flex flex-column">
        <TransformWrapper initialScale={1}>
          {!diagramError && <ZoomResetControls setRotation={setRotation} />}
          <TransformComponent>
            {!diagramError && (
              <Image
                fluid
                style={{
                  transform: `rotate(${rotation}deg)`,
                }}
                src={`${CR3_DIAGRAM_BASE_URL}/${crash.record_locator}.jpeg`}
                alt="crash diagram"
                onError={() => {
                  console.error("Error loading CR3 diagram image");
                  setDiagramError(true);
                }}
              />
            )}
```

The more time I have been spending with bootstrap, the more I find that flexboxes are usually going to solve most layout problems, especially for growing to fit and managing vertical and horizontal alignment. It's always a bunch of trial and error, but the [bootstrap docs](https://getbootstrap.com/docs/5.3/utilities/flex/) have some nice visual helpers.